PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:6 + add:0, change matches:
-    _inherit = 'loyalty.card'
-    _inherit = 'loyalty.history'
-    _inherit = 'loyalty.program'
-    _inherit = 'loyalty.reward'
-    _inherit = 'sale.order'
-    _inherit = 'sale.order.line'

Total Changes: 13

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/sale_loyalty/models/loyalty_card.py
+++ b/addons/sale_loyalty/models/loyalty_card.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class LoyaltyCard(models.Model):
-    _inherit = 'loyalty.card'
+    _inherit = ['loyalty.card']
 
     order_id = fields.Many2one(
         comodel_name='sale.order',

--- a/addons/sale_loyalty/models/loyalty_history.py
+++ b/addons/sale_loyalty/models/loyalty_history.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class LoyaltyHistory(models.Model):
-    _inherit = 'loyalty.history'
+    _inherit = ['loyalty.history']
 
     def _get_order_portal_url(self):
         if self.order_id and self.order_model == 'sale.order':

--- a/addons/sale_loyalty/models/loyalty_program.py
+++ b/addons/sale_loyalty/models/loyalty_program.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class LoyaltyProgram(models.Model):
-    _inherit = 'loyalty.program'
+    _inherit = ['loyalty.program']
 
     order_count = fields.Integer(compute='_compute_order_count')
     sale_ok = fields.Boolean(string="Sales", default=True)

--- a/addons/sale_loyalty/models/loyalty_reward.py
+++ b/addons/sale_loyalty/models/loyalty_reward.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class LoyaltyReward(models.Model):
-    _inherit = 'loyalty.reward'
+    _inherit = ['loyalty.reward']
 
     def _get_discount_product_values(self):
         res = super()._get_discount_product_values()

--- a/addons/sale_loyalty/models/sale_order.py
+++ b/addons/sale_loyalty/models/sale_order.py
@@ -17,7 +17,7 @@ def _generate_random_reward_code():
 
 
 class SaleOrder(models.Model):
-    _inherit = 'sale.order'
+    _inherit = ['sale.order']
 
     # Contains how much points should be given to a coupon upon validating the order
     applied_coupon_ids = fields.Many2many(

--- a/addons/sale_loyalty/models/sale_order_coupon_points.py
+++ b/addons/sale_loyalty/models/sale_order_coupon_points.py
@@ -5,7 +5,6 @@ from odoo import fields, models
 
 
 class SaleOrderCouponPoints(models.Model):
-    _name = 'sale.order.coupon.points'
     _description = 'Sale Order Coupon Points - Keeps track of how a sale order impacts a coupon'
 
     order_id = fields.Many2one(comodel_name='sale.order', required=True, ondelete='cascade')

--- a/addons/sale_loyalty/models/sale_order_line.py
+++ b/addons/sale_loyalty/models/sale_order_line.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class SaleOrderLine(models.Model):
-    _inherit = 'sale.order.line'
+    _inherit = ['sale.order.line']
 
     is_reward_line = fields.Boolean(
         string="Is a program reward line", compute='_compute_is_reward_line')
